packages/nuxt3/src/app/composables/router.ts

```diff
@@ -0,0 +1,65 @@
+import type { Router, RouteLocationNormalizedLoaded, NavigationGuard, RouteLocationNormalized, RouteLocationRaw } from 'vue-router'
+import { useNuxtApp } from '#app'
+
+export const useRouter = () => {
+  return useNuxtApp().$router as Router
```

Comment:
Just in case: 

```suggestion
  return useNuxtApp()?.$router as Router
```

Alternatively, we could introduce a flag to `useNuxtApp` to throw an error if not available but I believe this is more graceful behavior.

Reply:
So this would effectively be usage outside of setup. 👍

I'm okay with it, but it will likely also error immediately afterwards when user does `router.push()` or equivalent.